xc/device.ml

```diff
@@ -2787,11 +2787,28 @@ module Backend = struct
                       (string_of_message msg)
 
     let qmp_event_thread () =
-      debug "Starting QMP_Event thread";
-      (* Add the existing qmp sockets first *)
-      Sys.readdir var_run_xen_path
-      |> Array.to_list
-      |> List.iter (fun x -> try Scanf.sscanf x "qmp-event-%d" add with _ -> ());
```

Comment:
So `add` was already protected by an exception handler, but it didn't log the exception. The `Sys.readdir` call was not protected; is this the main fix here?

Reply:
The fix is:
* Separate exception handling from `sscanf` and `add` because we expect `sscanf` to raise exception that we can ignore but would want to log others.
* Make sure all exceptions in the setup phase are caught and reported because they would kill the thread